detekt-core/src/main/kotlin/io/gitlab/arturbosch/detekt/core/DetektFacade.kt

```diff
@@ -30,23 +28,19 @@ class DetektFacade(
     private val compiler = KtTreeCompiler.instance(settings)
 
     fun run(): Detektion {
-        val notifications = mutableListOf<Notification>()
-        val findings = HashMap<String, List<Finding>>()
-
         val filesToAnalyze = inputPaths.flatMap(compiler::compile)
         val bindingContext = generateBindingContext(filesToAnalyze)
 
         processors.forEach { it.onStart(filesToAnalyze) }
 
-        findings.mergeSmells(detektor.run(filesToAnalyze, bindingContext))
+        val findings = detektor.run(filesToAnalyze, bindingContext)
```

Comment:
```suggestion
        val findings: Map<String, List<Finding>> = detektor.run(filesToAnalyze, bindingContext)
```

It's not obvious what this function returns. (I'm not even sure if it is that what it returns)